src/runtime-rs/crates/hypervisor/README.md

```diff
@@ -89,6 +92,215 @@ pub trait Hypervisor: Send + Sync {
     async fn save_state(&self) -> Result<HypervisorState>;
    }
 ```
+### VM start procedure: steps through time
 In current design, VM will be started in the following steps.
 
 ![vmm start](../../docs/images/vm-start.svg)
+
+### Semantic of each function of the trait
+In runtime-rs, most functions' implementation are VMM-specific, since currently `Dragonball` is default
+VMM and implemented correctly, following section will explain two things.
+1. In high-level, what is the expected behavior of each function.
+2. What `Dragonball`(DBS) does in each function. This could be a reference, but remember that each
+VMM should have their own orientations.
+
+#### Explanations
+```rust
+async fn prepare_vm(&self, id: &str, netns: Option<String>) -> Result<()>
+```
+1. What is expected?
+* In general, `prepare_vm()` need the VMM to prepare everything **that VMM needs before boot**. 
+For example, maybe fill in some fields of VMM, or maybe create something on host that the VMM needs when booting. But only some preparations need to be done.
+
+
+2. What DBS does?
+* DBS fills in some of its fields, e.g. `id, state, jailer_root, vm_path, netns`, and create a Unix Domain Socket(uds) for communication. Note that the uds is not directly added to the vmm, but added to a buffer. The devices
```

Comment:
`buffer` is misleading, let people think about something like a memory buffer.Below you are using `some buffer(pending_list)`, this will be better for understanding.

Reply:
Fixed.